static/app/utils/useHasMounted.tsx

```diff
@@ -0,0 +1,14 @@
+import {useEffect, useRef} from 'react';
+/**
+ * Provides a boolean indicating if the component has completed its initial mount.
+ * @returns true if the component has mounted, false otherwise.
+ */
+export function useHasMounted() {
+  const hasMounted = useRef(false);
+
+  useEffect(() => {
+    hasMounted.current = true;
+  }, []);
```

Comment:
Should here also be a cleanup function that, otherwise on re-renders component might appear to be mounted, when it's not:

```
return () => { hasMounted.current = false }
```

Reply:
https://github.com/getsentry/sentry/pull/77469#discussion_r1758800968